docs/tutorials/custom_segmentation_trainer.py

```diff
@@ -0,0 +1,194 @@
+            "optimizer": optimizer,
+            "lr_scheduler": {"scheduler": scheduler, "monitor": self.monitor},
+        }
+
+    def configure_metrics(self) -> None:
+        """Initialize the performance metrics."""
+        num_classes: int = self.hparams["num_classes"]
+
+        self.train_metrics = MetricCollection(
+            {
+                "OverallAccuracy": Accuracy(
+                    task="multiclass", num_classes=num_classes, average="micro"
+                ),
+                "OverallPrecision": Precision(
+                    task="multiclass", num_classes=num_classes, average="micro"
+                ),
+                "OverallRecall": Recall(
+                    task="multiclass", num_classes=num_classes, average="micro"
+                ),
+                "OverallF1Score": FBetaScore(
+                    task="multiclass",
+                    num_classes=num_classes,
+                    beta=1.0,
+                    average="micro",
+                ),
+                "MeanIoU": JaccardIndex(
+                    num_classes=num_classes, task="multiclass", average="macro"
+                ),
+            },
+            prefix="train_",
+        )
+        self.val_metrics = self.train_metrics.clone(prefix="val_")
+        self.test_metrics = self.train_metrics.clone(prefix="test_")
+
+    def configure_callbacks(self):
+        """Initialize callbacks for saving the best and latest models.
+
+        Returns:
+            List of callbacks to apply.
+        """
+        return [
+            ModelCheckpoint(every_n_epochs=50, save_top_k=-1),
+            ModelCheckpoint(monitor=self.monitor, mode=self.mode, save_top_k=5),
+        ]
+
+    def on_train_epoch_start(self) -> None:
+        """Log the learning rate at the start of each training epoch."""
+        lr = self.optimizers().param_groups[0]["lr"]
+        self.logger.experiment.add_scalar("lr", lr, self.current_epoch)
+
+
+# ## Train model
+#
+# The remainder of the turial is straightforward and follows the typical [PyTorch Lightning](https://lightning.ai/) training routine. We instantiate a `DataModule` for the LandCover.AI dataset, instantiate a `CustomSemanticSegmentationTask` with a U-Net and ResNet-50 backbone, then train the model using a Lightning trainer.
+
+dm = LandCoverAIDataModule(root="data/", batch_size=64, num_workers=8, download=True)
+
+task = CustomSemanticSegmentationTask(
+    model="unet",
+    backbone="resnet50",
+    weights=True,
+    in_channels=3,
+    num_classes=6,
+    loss="ce",
+    lr=1e-3,
+    tmax=50,
+)
+
+# validate that the task's hyperparameters are as expected
+task.hparams
+
+# +
+accelerator = "gpu" if torch.cuda.is_available() else "cpu"
+
+trainer = pl.Trainer(
+    accelerator=accelerator,
+    devices=[0],
+    min_epochs=150,
+    max_epochs=300,
+    log_every_n_steps=50,
+)
+# -
+
+trainer.fit(task, dm)
+
+# ## Test model
+#
+# Finally, we test the model on the test set and visualize the results.
```

Comment:
`visualize the results` - would make me expect some plots, it is a table of metrics however

Reply:
Yep -- not finished here. Need to get some agreement on:
1. How to review
2. How to deal with long running notebooks in CI. If I run this notebook for 250 epochs (7 hours on my machine) then I get 85.95 test mIoU which is better than the results reported in https://arxiv.org/pdf/2005.02264.pdf. I think fake datasets are less interesting.
3. How to deal with the above issue I pointed out where the `ignore` parameter is being saved as a hparam, then breaking the subclass constructor.

Reply:
I think it would be fine to train for 1 batch as this notebook is about demonstrating the technique, in which case a CPU can be used